mipengine/controller/node_landscape_aggregator.py

```diff
@@ -199,47 +210,63 @@ def start(self):
     def stop(self):
         self.keep_updating = False
 
+    def set_registries(self, nodes, data_models, dataset_locations):
+        _node_registry = NodeRegistry(self._logger)
+        _node_registry.nodes = nodes
+        _data_model_registry = DataModelRegistry(self._logger)
+        _data_model_registry.data_models = data_models
+        _data_model_registry.dataset_locations = dataset_locations
+        self._nla_registries = _NLARegistries(
+            node_registry=_node_registry, data_model_registry=_data_model_registry
```

Comment:
All the registries could be immutable and the `nodes` , `data_models`, etc could be provided in the init.